### ISSUE TITLE
Cover replay attacks for attested build environments

### ISSUE THREAD

**Issue by marcelamelara** [slsa-framework/slsa#1167]
This is something that @paveliak and I have been discussing a little bit already, but don't have full fleshed out requirements yet. It seems particularly important to prevent/detect replay attacks of attested build environments since the goal is to start builds from a known good "clean" state.

Replay attacks also came up in https://github.com/slsa-framework/slsa/pull/1115#discussion_r1781390155.

**Comment by marcelamelara** [comment]
@paveliak I think this is something you're already covering in #1287 in the discussion on build environment reuse?

**Comment by paveliak** [comment]
@marcelamelara Yes, and I really like "replay attack" heading for it. I plan to adjust the wording to make the requirements less prescriptive. Presently we demand `build_id` to be included into the TPM quote. That works but equally it could be any globally unique identifier, e.g. if reimaging VM would produce a new unique Endorsement Key then it would equally serve the purpose.
